### PR TITLE
Create iOS_onboarding.toml

### PR DESCRIPTION
**Pull request by rhythmofrain63**
Added iOS_onboarding.toml - query for generating the underlying table in the iOS onboarding dashboard.

## Related Tickets & Documents
* DENG-XXXX
* DSRE-XXXX

**Reviewer, please follow [this checklist](https://github.com/mozilla/bigquery-etl/.github/reviewer_checklist.md)**

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-5341)

### CODE REVIEW

File: sql_generators/funnels/configs/iOS_onboarding.toml [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+        "fourth_card_secondary_click",
+        "fourth_card_close_click",
+        "fifth_card_impression",
+        "fifth_card_primary_click",
+        "fifth_card_secondary_click",
+        "fifth_card_close_click",
+         "sync_sign_in",
+         ]
+
+
```
Comment (kik-kik) [inline sql_generators/funnels/configs/iOS_onboarding.toml:37]: Do we need `[steps]` here?

For example I can see in another similar config this section being defined:
https://github.com/mozilla/bigquery-etl/blob/ece677485b8c30c0bbc92552851ed68fbabe2840/sql_generators/funnels/configs/email_first_reg_login_funnels_by_service.toml#L22

File: sql_generators/funnels/configs/iOS_onboarding.toml [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+aggregation = "count distinct"
+
+
+[data_sources]
+
+[data_sources.onboarding_events]
+from_expression = """
+  firefox_ios.firefox_ios_clients ic --each client_id has only one row
+  LEFT JOIN `moz-fx-data-shared-prod`.firefox_ios_derived.funnel_retention_clients_week_4_v1 r --each client_id has only one row
+  ON ic.client_id = r.client_id
```
Comment (kik-kik) [inline sql_generators/funnels/configs/iOS_onboarding.toml:247]: Looks like you could also use the more concise `USING(client_id)` here instead.

File: sql_generators/funnels/configs/iOS_onboarding.toml [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+    1 + LENGTH(ANY_VALUE(`mozfun.map.get_key`(event_extra, 'sequence_id'))) - 
+             LENGTH(REPLACE(ANY_VALUE(`mozfun.map.get_key`(event_extra, 'sequence_id')), '_', '')) AS number_of_onboarding_cards
+    FROM firefox_ios.events_unnested
+    WHERE `mozfun.map.get_key`(event_extra, 'sequence_id') IS NOT NULL
+    AND DATE(submission_timestamp) = @submission_date
+    GROUP BY 1) funnel_ids
+  ON ic.client_id  = funnel_ids.client_id
+
+"""
+submission_date_column = "ic.submission_date"
```
Comment (kik-kik) [inline sql_generators/funnels/configs/iOS_onboarding.toml:265]: I wonder if the `ic.` here is causing the dry run to fail.

File: sql_generators/funnels/configs/iOS_onboarding.toml
Comment (scholtzan) [inline sql_generators/funnels/configs/iOS_onboarding.toml:1]: I recommend lowercasing the filename here to stay consistent with the naming schema in this repo.

```suggestion
ios_onboarding.toml
```

**Review by scholtzan — COMMENTED** [review]
The `dry-run-sql` check is timing out since the generated query here is quite complex. I was able to run it without issues in BigQuery though (finished in 2+ minutes). So we could add an entry here to skip dry runs: https://github.com/mozilla/bigquery-etl/blob/f85d88c86f3adb2feafa91cf661c56cd9a1cbeec/bqetl_project.yaml#L170

Adding
```
  - sql/moz-fx-data-shared-prod/firefox_ios_derived/ios_onboarding_v1/query.sql
```
should do the trick

**Review by scholtzan — APPROVED** [review]
LGTM from the way the funnel is configured. Not sure if we want to wait on kik to review the logic of these queries?

**Comment by kik-kik** [comment]
I'm not familiar with the business process this is addressing so I'm not sure I'm the right person to validate the logic itself. Seeing all checks pass, from technical perspective I think we can merge this and perhaps it's best if @rhythmofrain63 validates the data after adding the config.

**Pull request merged**